tests/pytest/test.py

```diff
@@ -1114,6 +1115,46 @@ def testCopyCommand(env):
     r.assertEqual(json.loads(res), [new_values])
 
 
+def nest_object(depth, max_name_len, leaf_key, leaf_val):
+    """ Generate a JSON string such as {"a":{"b":{"c":{"leaf":42}}}} """
+    res = ""
+    for i in range(1, depth - 1):
+        name = ''.join([chr(random.randint(ord('a'), ord('z'))) for _ in range(0, random.randint(1, max_name_len))])
+        res = res + '{{"{}":'.format(name)
+    res = res + '{{"{}":{}'.format(leaf_key, leaf_val)
+    for i in range(1, depth):
+        res = res + '}'
+    return res
+
+def testNesting(env):
+    """ Test JSONPath Object nesting depth """
+    r = env
+
+    # Max nesting level for a single JSON value
+    depth = 128
+    doc = nest_object(depth, 5, "__leaf", 42)
+    r.assertOk(r.execute_command('JSON.SET', 'test', '$', doc))
+    res = r.execute_command('JSON.GET', 'test', '$..__leaf')
+    r.assertEqual(res, '[42]')
+
+    # No overall max nesting level (can exceeded the single value max nesting level)
+    doc = nest_object(depth, 5, "__deep_leaf", 420)
+    r.execute_command('JSON.SET', 'test', '$..__leaf', doc)
+    res = r.execute_command('JSON.GET', 'test', '$..__deep_leaf')
+    r.assertEqual(res, '[420]')
+
+    doc = nest_object(depth, 5, "__helms_deep_leaf", 42000)
```

Comment:
So we can create as deep nesting object as we want but not in a single `json.set` command? Is that right? Should we have some total limit as well (Maybe configurable)?

Reply:
Yes, the `test` doc will exceed the limit.
The limit is coming from `serde_json` when we deserialize the JSON value argument in the command.
Why limit if we don't have to? I suppose few users, if any, would get here.
Seems enough for me to document and add a test.

Reply:
Lets discuss this on another issue...